com.unity.ml-agents.extensions/Tests/Runtime/Sensors/CountingGridSensorTests.cs

```diff
@@ -0,0 +1,143 @@
+using System;
+using System.Collections;
+using System.Linq;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+using Unity.MLAgents.Sensors;
+using Unity.MLAgents.Extensions.Sensors;
+using Object = UnityEngine.Object;
+
+namespace Unity.MLAgents.Extensions.Tests.Sensors
+{
+    public class CountingGridSensorTests
+    {
+        GameObject testGo;
+        GameObject boxGo;
+        TestCountingGridSensorComponent gridSensorComponent;
+
+        // Use built-in tags
+        const string k_Tag1 = "Player";
+        const string k_Tag2 = "Respawn";
+
+        [UnitySetUp]
+        public IEnumerator SetupScene()
+        {
+            testGo = new GameObject("test");
+            testGo.transform.position = Vector3.zero;
+            gridSensorComponent = testGo.AddComponent<TestCountingGridSensorComponent>();
+
+            boxGo = new GameObject("block");
+            boxGo.tag = k_Tag1;
+            boxGo.transform.position = new Vector3(3f, 0f, 3f);
+            boxGo.AddComponent<BoxCollider>();
+
+            yield return null;
+        }
+
+        [TearDown]
+        public void ClearScene()
+        {
+            Object.DestroyImmediate(boxGo);
+            Object.DestroyImmediate(testGo);
+        }
+
+        public class TestCountingGridSensorComponent : GridSensorComponent
+        {
+            public void SetParameters(string[] detectableTags)
+            {
+                DetectableTags = detectableTags;
+                CellScale = new Vector3(1, 0.01f, 1);
+                GridSize = new Vector3Int(10, 1, 10);
+                ColliderMask = LayerMask.GetMask("Default");
+                RotateWithAgent = false;
+                CompressionType = SensorCompressionType.None;
+            }
+
+            protected override GridSensorBase[] GetGridSensors()
+            {
+                return new GridSensorBase[] {
+                    new CountingGridSensor(
+                        "TestSensor",
+                        CellScale,
+                        GridSize,
+                        DetectableTags,
+                        CompressionType) };
+            }
+        }
+
+        // Copied from GridSensorTests in main package
+        public static float[][] DuplicateArray(float[] array, int numCopies)
```

Comment:
Didn't find a good way to share utility methods between main package tests and extension tests

Reply:
you can have the extension package depend on an asmdef that has this method in it.  Then you can use it between both.